src/compile/mark/bar.ts

```diff
@@ -17,6 +17,10 @@ export namespace bar {
 
     const stack = model.stack();
```

Comment:
If you specify both `x` and `x2`, it shouldn’t be automatically stacked by default.

The question is what if user specify `stacked` property, and specify `x2` -- which are conflicting.  Which one of them get precedence.  I would say `x2` has stronger precedence because `x2` is closer to the output that we finally produce (If you think about it, stack also produces `x2` in Vega).
Basically, whenever `x2` appears, we never make `stacked` truthy.